bftengine/src/bftengine/KeyStore.cpp

```diff
@@ -153,6 +154,11 @@ void ClusterKeyStore::loadAllReplicasKeyStoresFromReservedPages() {
 
 std::optional<ReplicaKeyStore> ClusterKeyStore::loadReplicaKeyStoreFromReserevedPages(const uint16_t& repID) {
   reservedPages_.loadReservedPage(resPageOffset() + repID, buffer_.size(), buffer_.data());
+  if (buffer_.empty()) {
```

Comment:
you can check the `loadReservedPage()` status

Reply:
I think that's a good idea, thanks. For it to work in the test, we can change `ReservedPagesMock::loadReservedPage()` so that it returns `false` in case the passed length is 0.

However, in the actual implementation, if there is a pending reserved page, we will call `psd_->getPendingResPage()` with a size of zero and still return `true`:
https://github.com/vmware/concord-bft/blob/master/bftengine/src/bcstatetransfer/BCStateTran.cpp#L562

WDYT?

Reply:
Actually, as we've discussed offline, the `buffer_` is not supposed to be empty. I will add an assert when constructing the `ClusterKeyStore` instead. It is solely a test issue, it seems - we just need to pass a valid reserved page size instead of 0. Will change the PR accordingly.